src/cryptonote_basic/tx_extra.h

```diff
@@ -55,12 +55,25 @@
 #define TX_EXTRA_TAG_TX_KEY_IMAGE_UNLOCK        0x77
 #define TX_EXTRA_TAG_SERVICE_NODE_STATE_CHANGE  0x78
 #define TX_EXTRA_TAG_BURN                       0x79
+#define TX_EXTRA_TAG_LOKI_NAME_SYSTEM           0x7A
 
 #define TX_EXTRA_MYSTERIOUS_MINERGATE_TAG       0xDE
 
 #define TX_EXTRA_NONCE_PAYMENT_ID               0x00
 #define TX_EXTRA_NONCE_ENCRYPTED_PAYMENT_ID     0x01
 
+namespace lns
+{
+enum struct mapping_type : uint16_t
+{
+  messenger      = 0,
+  start_reserved = 1,
```

Comment:
Ah, okay in that case: shouldn't `start_reserved` and `blockchain` be the same (i.e. both `= 1`)?  Otherwise when we move `start_reserved` there is going to be a hole at value 1.

Reply:
Hmm. This is gonna be awkward at the upgrade period when we want to move the allowable range with the pre-hardfork needing to do type < unusable range - N then type <= unusable range for detecting valid types. This was only useful really for drafting up LNS

I've made this so that is more conventional to how we currently do it in the codebase with a hardfork aware function.

```c++
constexpr bool mapping_type_allowed(uint8_t /*hf_version*/, uint16_t type)
{
  return type == static_cast<uint16_t>(mapping_type::session);
}
constexpr bool mapping_type_allowed(uint8_t hf_version, mapping_type type)
{
  return mapping_type_allowed(hf_version, static_cast<uint16_t>(type));
}
```